src/test/suite/oidc.test.ts

```diff
@@ -0,0 +1,431 @@
+    };
+
+    assert.strictEqual(
+      await testConnectionController.addNewConnectionStringAndConnect(
+        connectionString
+      ),
+      true
+    );
+
+    const connectionId = testConnectionController.getActiveConnectionId();
+    if (!connectionId) {
+      assert.fail('Connection id not found for active connection');
+    }
+
+    await testConnectionController.disconnect();
+
+    assert.strictEqual(
+      await testConnectionController.connectWithConnectionId(connectionId),
+      true
+    );
+    assert.strictEqual(tokenFetchCalls, 2);
+  });
+
+  test('can cancel a connection attempt and then successfully connect', async function () {
+    const emitter = new EventEmitter();
+    const secondConnectionEstablished = once(
+      emitter,
+      'secondConnectionEstablished'
+    );
+    overrideRequestHandler = async (url) => {
+      if (new URL(url).pathname === '/authorize') {
+        emitter.emit('authorizeEndpointCalled');
+        // This does effectively mean that our 'fake browser'
+        // will never get a response from the authorization endpoint
+        // during the first connection attempt, and that therefore
+        // the local HTTP server will never have its redirect endpoint
+        // accessed.
+        await secondConnectionEstablished;
+      }
+    };
+
+    testConnectionController
+      .addNewConnectionStringAndConnect(connectionString)
+      .catch(() => {
+        // ignored
+      });
+
+    await once(emitter, 'authorizeEndpointCalled');
+    overrideRequestHandler = () => {};
+    const connected =
+      await testConnectionController.addNewConnectionStringAndConnect(
+        connectionString
+      );
+    emitter.emit('secondConnectionEstablished');
+    assert.strictEqual(connected, true);
+  });
+
+  test('can successfully re-authenticate', async function () {
+    showInformationMessageStub.resolves('Confirm');
+    const originalReAuthHandler =
+      testConnectionController._reauthenticationHandler.bind(
+        testConnectionController
+      );
+    let resolveReAuthPromise: (value?: unknown) => void;
+    const reAuthPromise = new Promise((resolve) => {
+      resolveReAuthPromise = resolve;
+    });
+    sandbox
+      .stub(testConnectionController, '_reauthenticationHandler')
+      .callsFake(async () => {
+        resolveReAuthPromise();
+        await originalReAuthHandler();
+      });
+    let tokenFetchCalls = 0;
+    let afterReauth = false;
+    getTokenPayload = () => {
+      tokenFetchCalls++;
+      return {
+        ...DEFAULT_TOKEN_PAYLOAD,
+        ...(afterReauth ? {} : { expires_in: 1 }),
+      };
+    };
+
+    const connected =
+      await testConnectionController.addNewConnectionStringAndConnect(
+        connectionString
+      );
+    assert.strictEqual(connected, true);
+    afterReauth = true;
+
+    // Wait for auth to expire, our auth expire in 1 second
+    await new Promise((resolve) => setTimeout(resolve, 1100));
```

Comment:
Can we make the 1 second less? Thinking how to make these tests faster. A bit to what Maurizio already mentioned, something we've been trying to move away from in our tests (VSCode unfortunately has a lot of them) is using setTimeout's to delay before a test condition. We prefer finding ways to use promise based listeners so things can resolve the moment they happen. Fine as is too.

Reply:
Yea I actually wanted to remove this as well but couldn't figure out a straight way to do that because here we are waiting for the OIDC token to expire which is marked to expire in a second. Maybe I will try to do some loops until an event is emitted or something like that. I will update here.

Reply:
Removed these in my recent commits